packages/backend-core/src/utils/utils.ts

```diff
@@ -69,6 +65,18 @@ export function isServingApp(ctx: Ctx) {
   return false
 }
 
+export function isServingBuilder(ctx: Ctx): boolean {
+  return ctx.path.startsWith(BUILDER_REFERER_PREFIX)
```

Comment:
Just to note some differences on the builder urls

- `/builder/app` - editing an app, including data, design, automation and any screens within
- `/builder/portal` - everywhere outside of an app, aside from authentication screens

We could possibly rename `BUILDER_REFERER_PREFIX` to `BUILDER_APP_PREFIX` to be more clear as the current constants looks very 'referrer' specific 


Background on the `BUILDER_REFERER_PREFIX` - it's only used for determining the `appId` in `getAppIdFromCtx`. This is needed when using the app builder as we don't have the app id anywhere else in the request at that point. We get it from the referrer of any requests sent when using the builder e.g. when fetching tables.